owncloudApp/src/main/java/com/owncloud/android/workers/CameraUploadsWorker.kt

```diff
@@ -0,0 +1,291 @@
+        syncType: SyncType,
+        numberOfFilesToUpload: Int
+    ) {
+        val contentText = when (syncType) {
+            SyncType.PICTURE_UPLOADS -> R.string.uploader_upload_picture_upload_files
+            SyncType.VIDEO_UPLOADS -> R.string.uploader_upload_video_upload_files
+        }
+
+        NotificationUtils.createBasicNotification(
+            context = appContext,
+            contentTitle = appContext.getString(R.string.uploader_upload_camera_upload_files),
+            contentText = appContext.getString(contentText, numberOfFilesToUpload),
+            notificationChannelId = UPLOAD_NOTIFICATION_CHANNEL_ID,
+            notificationId = syncType.getNotificationId(),
+            intent = NotificationUtils.composePendingIntentToUploadList(appContext),
+            onGoing = false,
+            timeOut = 5_000
+        )
+    }
+
+    private fun showNotificationToUpdateUri(
+        syncType: SyncType
+    ) {
+        val contentText: Int = when (syncType) {
+            SyncType.PICTURE_UPLOADS -> R.string.uploader_upload_picture_upload_error
+            SyncType.VIDEO_UPLOADS -> R.string.uploader_upload_video_upload_error
+        }
+        NotificationUtils.createBasicNotification(
+            context = appContext,
+            contentTitle = appContext.getString(R.string.uploader_upload_camera_upload_source_path_error),
+            contentText = appContext.getString(contentText),
+            notificationChannelId = UPLOAD_NOTIFICATION_CHANNEL_ID,
+            notificationId = syncType.getNotificationId(),
+            intent = NotificationUtils.composePendingIntentToUploadList(appContext),
+            onGoing = false,
+            timeOut = null
+        )
+    }
+
+    private fun updateTimestamp(folderBackUpConfiguration: FolderBackUpConfiguration, syncType: SyncType) {
+        val currentTimestamp = System.currentTimeMillis()
+
+        when (syncType) {
+            SyncType.PICTURE_UPLOADS -> {
+                val savePictureUploadsConfigurationUseCase: SavePictureUploadsConfigurationUseCase by inject()
+                savePictureUploadsConfigurationUseCase.execute(
+                    SavePictureUploadsConfigurationUseCase.Params(folderBackUpConfiguration.copy(lastSyncTimestamp = currentTimestamp))
+                )
+            }
+            SyncType.VIDEO_UPLOADS -> {
+                val saveVideoUploadsConfigurationUseCase: SaveVideoUploadsConfigurationUseCase by inject()
+                saveVideoUploadsConfigurationUseCase.execute(
+                    SaveVideoUploadsConfigurationUseCase.Params(folderBackUpConfiguration.copy(lastSyncTimestamp = currentTimestamp))
+                )
+            }
+        }
+    }
+
+    private fun getFilesReadyToUpload(
+        syncType: SyncType,
+        sourcePath: String,
+        lastSyncTimestamp: Long,
+    ): List<DocumentFile> {
+        val simpleDateFormat = SimpleDateFormat("yyyy/MM/dd HH:mm:ss", Locale.getDefault())
```

Comment:
Maybe it is better to place it as a constant

Reply:
I will remove it directly, we can use the Date().toString() 💯